### ISSUE TITLE
feat(model): route streamed Q4_K mmap spans into Metal

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#9073]
## Parent context

#8325
Depends on: #9072
Related shipped primitive: #8969

## Outcome

A streamed dense Q4_K tensor loaded from an mmap-backed GGUF shard reaches Metal through the offset-aware no-copy span path, never performs the aligned heap `ReadAt` promotion, and releases every borrowing Q4_K handle before the checkpoint mapping is closed.

## Current state

`WeightSource` already retains `data` and `dataFor` for mmap-backed GGUF shards, but `TensorBytes` deliberately copies and `lazyDenseQ4KTensorWork` reduces each streamed tensor to `io.ReaderAt + FileOffset + Bytes`. `q4kTensor.materializeRaw` then allocates page-aligned heap backing and reads the whole payload on first Metal use. Model teardown releases mapped Q8 residency before `WeightSource.Close`, but it has no model-scoped Q4_K release step. #8969 proved a retained page-aligned mapped-range owner; #9072 owns the Metal base-plus-logical-offset upload contract.

## Why this is next

Once #9072 lands, the only remaining mechanism gap under #8325 is preserving a bounded mapped view from GGUF selection through lazy model ownership and ordering Q4_K handle release before unmap. This leaf closes that mechanism without claiming the parent's live startup/RSS win.

## Working spine

Select an eligible mapped tensor span from the tensor's actual shard -> attach the aligned base, bounded logical offset, and caller lifetime to `LazyQ4KRange` -> prefer #9072's mapped upload in `metalQ4KWeight` -> fall back to the current aligned `ReadAt` owner when no valid mapping exists -> release this model's cached Q4_K handles -> close the retained `WeightSource`.

## Core through-line

For local Mac operators running fak-native streamed Qwen Q4_K, preserve checkpoint-page ownership all the way into Metal; today the loader retains the map but throws away its view before the model can use it.

## Value

For: local Apple-Silicon operators serving streamed fak-native Qwen Q4_K models.
Problem: an existing GGUF mmap is reduced to ReaderAt and copied into aligned heap backing on first Q4_K use.
Today: map ownership, lazy tensor description, Metal upload, and teardown do not carry one shared lifetime contract.
Better because: the checkpoint's immutable pages become the Q4_K Metal backing while portable and invalid spans retain the proven copy fallback.
Witness: one focused integration-test command proves mapped selection, zero ReaderAt promotion, exact native output, fallback, and Q4-before-unmap teardown.

## Problem frame

Centrality: Enabling - this is the loader/model integration beneath #8325's exact-model outcome.
P1: Preserved - quant bytes, tensor layouts, kernels, logits, and stop behavior stay unchanged.
P2: Advanced - the intermediate heap promotion is absent for eligible mapped spans.
P3: Preserved - opt-in mmap selection and the current aligned-reader fallback remain explicit.
P4: Advanced - selected path, fallback reason, per-model handle release, and unmap ordering become testable.

## Scope

Expose one bounds-checked page-aligned tensor-span view from the tensor's retained mmap shard; thread that optional view through streamed dense Q4_K construction; select #9072's upload without calling `materializeRaw`; preserve the existing ReaderAt fallback; and release only this model's cached Q4_K handles before its retained checkpoint closer runs.

## Out of scope

Do not change non-streamed tensors, non-Q4_K formats, mmap defaults, worker concurrency, admission bounds, Metal kernels, #9072's external-span API, global mapping caches, exact-model benchmark thresholds, or the parent keep/reject decision. Do not close #8325 from fixture-only evidence.

## Done condition

One streamed dense Q4_K tensor from a retained mmap shard carries a valid aligned mapped view into the offset-aware Metal upload, performs zero payload `ReadAt` calls on that path, produces CPU-reference-identical results, falls back byte-identically when mmap or span alignment is unavailable, and on `Model.CloseWeights` releases that model's borrowing Q4_K handles before closing/unmapping the checkpoint exactly once.

## Witness

`go test ./internal/ggufload ./internal/model -run '^Test(StreamedDenseQ4KMappedMetal|CloseWeightsReleasesMappedQ4K)' -count=1` uses a tiny non-page-aligned GGUF tensor fixture and an ordered closer to prove mapped no-copy selection, zero ReaderAt promotion, exact GEMV output, fallback behavior, active-session deferral, and Q4-handle release before unmap.

## Acceptance gate

The focused integration witness and existing streamed-Q4_K/mmap/Metal lifecycle suites pass; `fak validate --mine` is green for every listed path; a source audit proves the mapped path cannot outlive its `WeightSource` and the fallback cannot retain a borrowed slice.

## Dependencies

- #9072 must provide the offset-aware external-span Metal upload.
- #8969 is the shipped mapped-range lifetime reference; reuse its invariants rather than inventing weaker ownership.
- This issue lands before #8325's exact-checkpoint keep/reject campaign.
- It is independent of #9059's worker-count experiment and #9069's KV admission geometry.

## Closure binding

Close only from one ship commit carrying `Closes #ISSUE`, the captured focused command, explicit path validation, and the `(fak model)` trailer. A loader-only alias or a Metal upload without ordered teardown does not close it.

## Likely files

- `internal/ggufload/gguf_weightsource.go`
- `internal/ggufload/gguf_mmap_test.go`
- `internal/ggufload/quant_q4k_loader.go`
- `internal/ggufload/quant_q4k_loader_test.go`
- `internal/model/quant_q4k.go`
- `internal/model/quant_q4k_lazy_test.go`
- `internal/model/metal_q4k_on.go`
- `internal/model/metal_q4k_test.go`

## Lane

model

## Work unit

leaf

## Expected steps

5

## Work scale

S1

## Required model tier

tier/T2-required

## Optimal model tier

tier/T1-optimal

## Assumptions

- The retained shard mapping can expose the pages covering a tensor without permitting reads outside the mapped file.
- #9072 can wrap that aligned span while the model-owned `WeightSource` remains alive.
- Per-model `Q4KWeight.Release` is sufficient before unmap; no process-global reset is required.

## Confusion risks

- `ReaderAt` backed by mmap still copies and is not the desired path.
- A logical tensor slice alone does not expose the page-aligned base required by Metal.
- Releasing the checkpoint before its Metal handles is a use-after-unmap even if no Go slice remains.
- Passing this fixture does not establish #8325's startup or RSS gain.

## Coordination

Own the listed `ggufload` and `model` files under one model-lane lease. Do not overlap #9072's `internal/metalgemm/**` lease, #9069's context-size files, or #9059's loader-worker/modelbench files.

## Trigger

Dispatch only after #9072 is witnessed on trunk and the model lane is free.

## Batch policy

One end-to-end selection/lifetime integration and one composite witness. Keep live exact-checkpoint measurement in #8325.

## Placement

Generation: gen/next. Priority: P2. Milestone: Model support and routing parity. Lane: model. Completion standard: development.

## Generation

G1

## Target operating envelope

- retained-mmap-shards: >= 1 shard
- streamed-dense-q4k-paths: >= 1 path
- model-weight-closer-paths: >= 1 path

## Witnessed operating envelope

- retained-mmap-shards: >= 1 shard
- streamed-dense-q4k-paths: >= 1 path
- model-weight-closer-paths: >= 1 path

## Priority context

Promotion evidence: #8969 is shipped, the GGUF mmap view already exists, and heap promotion is explicit in `materializeRaw`. Demotion evidence: #9072 rejects valid GGUF offsets or the retained mapping cannot meet Metal's lifetime contract. Invalidating assumption: exact mapped bytes cannot safely back the registered buffer. Orthogonality: #8325 owns live startup/RSS evaluation; #9059 owns loader worker count.

## Gold-plating boundary

One Q4_K streamed-dense path, one fallback, and ordered per-model teardown. No cross-format zero-copy framework, mapping cache, eager all-weight upload policy, admission change, or performance number.

## Completion standard

development

## Work estimate

Estimate: 1 point. Uncertainty: the final partial file page may require a typed fallback rather than widening the mapping beyond its safe slice.

## Overall completion contribution

Parent #8325 scope baseline: 3 points. Contribution: 1/3 points for loader/model integration; the final 1/3 remains #8325's exact-model keep/reject campaign.
